src/sst/elements/ariel/arielcore.cc

```diff
@@ -901,7 +901,7 @@ void ArielCore::tick() {
         currentCycles++;
         statCycles->addData(1);
 
-        if( updateCycle ) {
+        if( updateCycle || isStalled) {
```

Comment:
Why do you need to update activeCycles if the core is stalled?

Reply:
For example if the core is waiting for the completion of shootdown event, the core is stalled but the cycle needs to be counted as cycles are wasted in such scenarios. So cycles need to updated while the core is stalled.

Reply:
Current cycles is updated, which corresponds to the stat cycles. statActiveCycles corresponds to the stat active_cycles. Active cycles are only updated when events are processed -- the stall flag prevents new events from being issued.

Reply:
@hughes-c Can I use a new stat variable(stalled_cycles) which just counts the number of cycles for which the core is stalled?